tests/functional/src/main/java/org/mule/functional/junit4/runners/ArtifactClassloaderTestRunner.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.functional.junit4.runners;
+
+import static java.util.Collections.emptyList;
+import static org.mule.runtime.core.util.ClassUtils.withContextClassLoader;
+import org.mule.runtime.module.artifact.classloader.ArtifactClassLoader;
+
+import java.lang.annotation.Annotation;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.runner.Description;
+import org.junit.runner.Runner;
+import org.junit.runner.notification.RunNotifier;
+import org.junit.runners.BlockJUnit4ClassRunner;
+import org.junit.runners.Suite;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.InitializationError;
+import org.junit.runners.model.TestClass;
+
+/**
+ * Runner that does the testing of the class using a different {@link ClassLoader} from the one that launched the test.
+ *
+ * @since 4.0
+ */
+public class ArtifactClassloaderTestRunner extends Suite
```

Comment:
It does not seem right to me to extend Suite instead of just implementing a runner and delegating to another one (the default or the configured using @RunnerDelegateTo)